src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -364,9 +360,17 @@ void ImageReplayer<I>::bootstrap() {
 
   request->get();
   m_bootstrap_request = request;
+
+  // If the ImageReplayer is stopped before the bootstrap, proceed to 
+  // prepare the remote image in order to delete it if the primary image is deleted.
```

Comment:
Somewhat nit-picking but please bear with me ;)  `ImageReplayer` is not stopped here (this code wouldn't be executing if it were) and "delete it" is easy to interpret as "delete the remote image".  How about something like:
```
  // proceed even if stop was requested to allow for m_delete_requested
  // to get set; cancel() would prevent BootstrapRequest from going into
  // image sync
```

Reply:
Done.